src/svm/utils/report.c

```diff
@@ -0,0 +1,190 @@
+{
+  PetscInt  confusion_mat[4];
+  PetscInt  TP,FP,TN,FN;
+
+  PetscReal accuracy[2];
+  PetscReal auc_roc;
+  PetscReal precision[3];
+  PetscReal sensitivity[3];
+  PetscReal F1[3];
+  PetscReal jaccard_index[3];
+
+  PetscReal specifity;
+  PetscReal TPR,FPR;
+  PetscReal x[3],y[3],dx;
+
+  PetscInt  i;
+
+  PetscFunctionBegin;
+  PetscCall(BinaryConfusionMatrix(svm,y_pred,y_known,confusion_mat));
+
+  TP = confusion_mat[0];
+  FP = confusion_mat[1];
+  FN = confusion_mat[2];
+  TN = confusion_mat[3];
+
+  if (cmat != NULL) {
+    PetscCall(PetscMemcpy(cmat,confusion_mat,4 * sizeof(PetscInt)));
+  }
+
+  if (scores == NULL) {
+    PetscFunctionReturn(PETSC_SUCCESS);
+  }
+
+  precision[0] = (PetscReal) TN / (PetscReal) (TN + FP); /* precision (negative class) */
+  precision[1] = (PetscReal) TP / (PetscReal) (TP + FN); /* precision (positive class) */
+  precision[2] = (precision[0] + precision[1]) / 2.;     /* precision (average)        */
+
+  sensitivity[0] = (PetscReal) TN / (PetscReal) (TN + FN); /* sensitivity (negative class) */
+  sensitivity[1] = (PetscReal) TP / (PetscReal) (TP + FP); /* sensitivity (positive class) */
+  sensitivity[2] = (sensitivity[0] + sensitivity[1]) / 2.; /* sensitivity (average)        */
+
+  F1[0] = 2. * (precision[0] * sensitivity[0]) / (precision[0] + sensitivity[0]); /* F1 (negative class) */
+  F1[1] = 2. * (precision[1] * sensitivity[1]) / (precision[1] + sensitivity[1]); /* F1 (positive class) */
+  F1[2] = (F1[0] + F1[1]) / 2.;                                                   /* F1 (average) */
+
+  jaccard_index[0] = (PetscReal) TN / (PetscReal) (TN + FN + FP); /* Jaccard index (negative class) */
+  jaccard_index[1] = (PetscReal) TP / (PetscReal) (TP + FP + FN); /* Jaccard index (positive class) */
+  jaccard_index[2] = (jaccard_index[0] + jaccard_index[1]) / 2.;  /* Jaccard index (average) */
+
+  /* Area under curve (trapezoidal rule) */
+  specifity = (PetscReal) TN / (PetscReal) (TN + FP);
+
+  FPR = 1 - specifity;
+  TPR = sensitivity[0];
+
+  x[0] = 0; x[1] = FPR; x[2] = 1;
+  y[0] = 0; y[1] = TPR; y[2] = 1;
+
+  auc_roc = 0.;
+  for (i = 0; i < 2; ++i) {
+    dx = x[i+1] - x[i];
+    auc_roc += ((y[i] + y[i+1]) / 2.) * dx;
+  }
+
+  /* Accuracy */
+  accuracy[0] = (PetscReal) (TP + TN) / (PetscReal) (TP + TN + FP + FN);
+
+  /* Set scores */
+  scores[0] = accuracy[0];
+
+  scores[1] = precision[0];
+  scores[2] = precision[1];
+  scores[3] = precision[2];
+
+  scores[4] = sensitivity[0];
+  scores[5] = sensitivity[1];
+  scores[6] = sensitivity[2];
+
+  scores[7] = F1[0];
+  scores[8] = F1[1];
+  scores[9] = F1[2];
+
+  scores[10] = jaccard_index[0];
+  scores[11] = jaccard_index[1];
+  scores[12] = jaccard_index[2];
+
+  scores[13] = -1.;
+  scores[14] = -1.;
+  scores[15] = auc_roc;
+
+  PetscFunctionReturn(PETSC_SUCCESS);
+}
+
+PetscErrorCode SVMPrintBinaryClassificationReport(SVM svm,PetscInt *cmat,PetscReal *scores,PetscViewer v)
```

Comment:
```suggestion
PetscErrorCode SVMBinaryViewClassificationReport(SVM svm,PetscInt *cmat,PetscReal *scores,PetscViewer v)
```
Why aren't these in `svm/impls/binary/`?